data/service/api/v1/devicetokens.go

```diff
@@ -0,0 +1,45 @@
+package v1
+
+import (
+	data "github.com/tidepool-org/platform/data/service"
+	"github.com/tidepool-org/platform/devicetokens"
+	"github.com/tidepool-org/platform/request"
+	platform "github.com/tidepool-org/platform/service"
+	"github.com/tidepool-org/platform/service/api"
+)
+
+func DeviceTokensRoutes() []data.Route {
+	return []data.Route{
+		data.MakeRoute("POST", "/v1/device-tokens/:userID", UpsertDeviceToken, api.RequireAuth),
```

Comment:
The (main) difference is that PUT is supposed to [guarantee idempotence](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT). I don't have a requirement for idempotence, nor do I want to guarantee it (as this might in the future trigger a test message) so I stuck with POST.

Reply:
Your choice.